### PR TITLE
feature(TDD): Add TDD setup and soft-enable 50% coverage markers 

### LINKED ISSUE #492: TDD setup
We should have a better setup for doing development with running the handful of tests relevant to the current feature/bug

Once this is closed, we should work on:
- #493

### PR DESCRIPTION
**Pull request by ankushdharkar**
Closes #492 

A developer can now set the files (with wildcards) and run `yarn tdd:watch` and make all the tests pass.

Also, we should make sure that our coverage from `yarn test` reaches a minimum of 50%, before enabling the flag

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,29 +16,50 @@
 ## About the Project
 This Project serves the backend APIs required for [Real Dev Squad](https://realdevsquad.com/) web projects. This project is built in [Express.js](https://expressjs.com/).
 
-## Running the Project
-We are moving to yarn, to ensure that we use yarn , we are moving ahead with [Volta](https://docs.volta.sh/guide/#why-volta)
+## Prerequisites
+- The application uses [node-config](https://github.com/lorenwest/node-config)([documentation](https://github.com/lorenwest/node-config/wiki/Configuration-Files)) for managing config.
+- Create a new file: `config/local.js`. Override the required config values from `config/development.js` and `config/default.js` into `config/local.js`.
```
Comment (swarajpure) [inline README.md:21]: Can we simply this even further? Override something from 2 files doesn't seem very beginner-friendly I guess? 🤔

File: README.md [hunk 2]
```diff
@@ -16,29 +16,50 @@
...
+- Register the application for [GitHub OAuth](https://docs.github.com/en/developers/apps/creating-an-oauth-app) to get the `clientId` and `clientSecret`. Add the callback URL as `http://<HOSTNAME>:<PORT>/auth/github/callback`
```
Comment (swarajpure) [inline README.md:22]: Can we add the example of callback URL here? Someone might not know what is HOSTNAME

File: scripts/tests/tdd.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# "tdd-files-list.txt only contain those files that are to be tested locally during development.
+# Wildcards are allowed in the filepaths. You cna add any number of paths.
```
Comment (swarajpure) [inline scripts/tests/tdd.sh:4]: "can"

File: README.md (hunk 1 above)
Comment (ankushdharkar) [reply README.md:21]: I think #500 will take care of this.

File: scripts/tests/tdd.sh (hunk 3 above)
Comment (ankushdharkar) [reply scripts/tests/tdd.sh:4]: Oops

Comment (ankushdharkar) [reply scripts/tests/tdd.sh:4]: Fixed in a new commit.

File: README.md (hunk 2 above)
Comment (ankushdharkar) [reply README.md:22]: This seems to be out of scope for this PR. It should be tackled in:
- #500

**Review by prakashchoudhary07 — APPROVED** [review]
